src/autoapp/Projection/GSTVideoOutput.cpp

```diff
@@ -0,0 +1,260 @@
+        "omxh264dec";
+        #endif
+    #endif
+    
+    
+    vid_pipeline = gst_parse_launch(vid_launch_str, &error);
+    bus = gst_pipeline_get_bus(GST_PIPELINE(vid_pipeline));
+    gst_bus_add_watch(bus, (GstBusFunc) GSTVideoOutput::bus_callback, this);
+    gst_object_unref(bus);
+
+    
+    
+
+    GstElement *sink = QGlib::RefPointer<QGst::Element>(m_videoSink);
+    // g_object_set (sink, "force-aspect-ratio", true, nullptr);
+    g_object_set (sink, "force-aspect-ratio", false, nullptr);
+
+    g_object_set (sink, "sync", false, nullptr);
+    g_object_set (sink, "async", false, nullptr);
+    GstElement *capsfilter = gst_bin_get_by_name(GST_BIN(vid_pipeline), "mycapsfilter");
+    gst_bin_add(GST_BIN(vid_pipeline), GST_ELEMENT(sink));
+    gst_element_link(capsfilter, GST_ELEMENT(sink));
+
+    vid_src = GST_APP_SRC(gst_bin_get_by_name(GST_BIN(vid_pipeline), "mysrc"));
+    gst_app_src_set_stream_type(vid_src, GST_APP_STREAM_TYPE_STREAM);
+
+    connect(this, &GSTVideoOutput::startPlayback, this, &GSTVideoOutput::onStartPlayback, Qt::QueuedConnection);
+    connect(this, &GSTVideoOutput::stopPlayback, this, &GSTVideoOutput::onStopPlayback, Qt::QueuedConnection);
+
+}
+
+GSTVideoOutput::~GSTVideoOutput() {
+    gst_object_unref(vid_pipeline);
+    gst_object_unref(vid_src);
+}
+
+gboolean GSTVideoOutput::bus_callback(GstBus */* unused*/, GstMessage *message, gpointer *ptr) {
+    gchar *debug;
+    GError *err;
+    gchar *name;
+
+    switch (GST_MESSAGE_TYPE(message)) {
+
+    case GST_MESSAGE_ERROR:
+        gst_message_parse_error(message, &err, &debug);
+        OPENAUTO_LOG(info) << "[GSTVideoOutput] Error "<< err->message;
+        g_error_free(err);
+        g_free(debug);
+        break;
+
+    case GST_MESSAGE_WARNING:
+        gst_message_parse_warning(message, &err, &debug);
+        OPENAUTO_LOG(info) << "[GSTVideoOutput] Warning "<<err->message<<" | Debug "<< debug;
+
+        name = (gchar *) GST_MESSAGE_SRC_NAME(message);
+
+        OPENAUTO_LOG(info) << "[GSTVideoOutput] Name of src "<< name ? name : "nil";
+        g_error_free(err);
+        g_free(debug);
+
+        break;
+
+    case GST_MESSAGE_EOS:
+        OPENAUTO_LOG(info) << "[GSTVideoOutput] End of stream";
+        break;
+
+    case GST_MESSAGE_STATE_CHANGED:
+        break;
+    default:
+        break;
+    }
+
+    return TRUE;
+}
+bool GSTVideoOutput::open()
+{
+     GstElement *capsfilter = gst_bin_get_by_name(GST_BIN(vid_pipeline), "mycapsfilter");
+    GstPad *convert_pad = gst_element_get_static_pad(capsfilter, "sink");
+    gst_pad_add_probe (convert_pad,GST_PAD_PROBE_TYPE_EVENT_DOWNSTREAM,convert_probe, this, NULL);
+    gst_element_set_state(vid_pipeline, GST_STATE_PLAYING);
+    return true;
+}
+
+GstPadProbeReturn GSTVideoOutput::convert_probe(GstPad *pad, GstPadProbeInfo *info, void *user_data){
+    GstEvent *event = GST_PAD_PROBE_INFO_EVENT(info);
+    if (GST_PAD_PROBE_INFO_TYPE(info) & GST_PAD_PROBE_TYPE_EVENT_DOWNSTREAM) {
```

Comment:
follow the style from the rest of the repo (bracket on newline after if/else) (also not a fan of this but gotta stay consistent)